vllm_spyre/compat_utils.py

```diff
@@ -1,5 +1,20 @@
+import inspect
 from dataclasses import fields
+from typing import Callable
 
 
 def dataclass_fields(cls: type) -> list[str]:
     return [f.name for f in fields(cls)]
+
+
+def has_argument(func: Callable, param_name: str) -> bool:
+    # Checks the signature of a method and returns true iff the method accepts
+    # a parameter named `$param_name`. This is slow and should not be used in
+    # the critical path.
```

Comment:
Reading this gave me a thought: can we track this and check to make sure we don't put this in a critical path? Maybe we track the rate of calls to this function or something and WARN if it exceeds a threshold.

Or probably better: use an `@lru_cache` on this and other compatibility functions that use `inspect`.